src/test/scala/chiselTests/PrintableSpec.scala

```diff
@@ -261,4 +254,134 @@ class PrintableSpec extends AnyFlatSpec with Matchers {
       """printf(clock, UInt<1>("h1"), "adieu AnonymousBundle(foo -> %d, bar -> %d)", myBun.foo, myBun.bar) : farewell"""
     )
   }
+
+  // Unit tests for cf
+  it should "print regular scala variables with cf format specifier" in {
+
+    class MyModule extends BasicTester {
+      val f1 = 20.45156
+      val i1 = 10
+      val s1: Short = 15
+      val l1: Long = 253
+      val str1 = "Printable String!"
+      printf(
+        cf"F1 = $f1 D1 = $i1 F1 formatted = $f1%2.2f s1 = $s1 l1 = $l1 str1 = $str1%s i1_string = $i1%s i1_hex=$i1%x"
+      )
+
+    }
+    generateAndCheck(new MyModule) {
+      case Seq(
+            Printf(
+              "F1 = 20.45156 D1 = 10 F1 formatted = 20.45 s1 = 15 l1 = 253 str1 = Printable String! i1_string = 10 i1_hex=a",
```

Comment:
Can you create and assign to a val this `Printf` object on an earlier line so that scalafmt doesn't butcher this so much?

Reply:
@jackkoenig  what's the heuristic which scalafmt uses to split? I did assign it to a val like this 

```
val expected = Printf("F1 = 20.4517 D1 = 10 F1 formatted = 20.45 s1 = 15 l1 = 253 str1 = Printable String! i1_string = 10 i1_hex=a",Seq())
    generateAndCheck(new MyModule) {
      case Seq(`expected`) =>
    }
```

But still it split it up like this 

```
val expected = Printf(
      "F1 = 20.4517 D1 = 10 F1 formatted = 20.45 s1 = 15 l1 = 253 str1 = Printable String! i1_string = 10 i1_hex=a",
      Seq()
    )
    generateAndCheck(new MyModule) {
      case Seq(`expected`) =>
    }
```

Reply:
You could do something like:
```scala
val msg = "F1 = 20.45156 D1 = 10 F1 formatted = 20.45 s1 = 15 l1 = 253 str1 = Printable String! i1_string = 10 i1_hex=a"
generateAndCheck(new MyModule) {
  case Seq(Printf(`msg`), Seq()) =>
}
```

some folks don't like the backticks for comparison instead of binding a value name so you could do:
```scala
val expected = "F1 = 20.45156 D1 = 10 F1 formatted = 20.45 s1 = 15 l1 = 253 str1 = Printable String! i1_string = 10 i1_hex=a"
generateAndCheck(new MyModule) {
  case Seq(Printf(msg), Seq()) if msg == expected =>
}
```

Reply:
@jackkoenig  - I took a reasonable shortcut - pruned the expected string to remove long and short types - did not add much value in terms of coverage - and that seems to have done the trick too. Hope that's fine - else can resort to your suggestion.